src/Sulu/Bundle/ContentBundle/Controller/NodeResourcelocatorController.php

```diff
@@ -39,13 +38,8 @@ public function postGenerateAction()
         $uuid = $this->getRequestParameter($this->getRequest(), 'uuid');
         $parts = $this->getRequestParameter($this->getRequest(), 'parts', true);
         $templateKey = $this->getRequestParameter($this->getRequest(), 'template', true);
-
-        list($webspaceKey, $languageCode) = $this->getWebspaceAndLanguage();
-        if ($templateKey === null) {
-            $webspaceManager = $this->container->get('sulu_core.webspace.webspace_manager');
-            $webspace = $webspaceManager->findWebspaceByKey($webspaceKey);
-            $templateKey = $webspace->getTheme()->getDefaultTemplate(Structure::TYPE_PAGE);
-        }
+        $webspaceKey = $this->getRequestParameter($this->getRequest(), 'webspace', true);
```

Comment:
the template key is mandatory and the function `getWebspaceAndLanguage` does this two function calls.